src/transformers/modeling_tf_utils.py

```diff
@@ -896,6 +1034,89 @@ def _linear(self, inputs):
         return tf.reshape(logits, first_dims + [self.vocab_size])
 
 
+class WordEmbeddings(tf.keras.layers.Layer):
```

Comment:
With a strict interpretation of the rule: "The full forward pass of the model should be in the same file.", those three classes should be in `modeling_tf_bert`. Not sure if we want to apply it to the letter here or not.

Reply:
Here these layers are very generic cases like the `TFSequenceSummary` or the `TFSharedEmbeddings` even in a broader perspective they won't be updated for one model or another and they must stay as they are (no exceptions, no if, etc...). So I would be in favor to keep them in a generic space.